server/src/test/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexServiceTests.java

```diff
@@ -1096,7 +1097,16 @@ public void testBuildIndexMetadata() {
 
         Settings indexSettings = indexSettings(IndexVersion.current(), 1, 0).build();
         List<AliasMetadata> aliases = List.of(AliasMetadata.builder("alias1").build());
-        IndexMetadata indexMetadata = buildIndexMetadata("test", aliases, () -> null, indexSettings, 4, sourceIndexMetadata, false);
+        IndexMetadata indexMetadata = buildIndexMetadata(
+            "test",
+            aliases,
+            () -> null,
+            indexSettings,
+            4,
+            sourceIndexMetadata,
+            false,
+            TransportVersion.current()
```

Comment:
Can we also test with an older minimum version? As well as add assertions on the expected eventIngested range?

Reply:
New test added in this commit: https://github.com/elastic/elasticsearch/pull/106252/commits/282170de4c5943c3bed71029ead70fccb4609f36

Let me know if that's not what you had in mind.

Reply:
That looks right to me, but I'd like to have an assertion in the test on the expected event.ingest range, which I think should be `UNKNOWN` for older transport versions and `NO_SHARDS` for newer?

Reply:
Yes. I will fix that in next push.